kolibri/plugins/management/assets/src/vue/manage-class-page/class-delete-modal.vue

```diff
@@ -68,14 +62,10 @@
         required: true,
       },
     },
-    data() {
-      return {
-        error_message: '',
-      };
-    },
     methods: {
-      submit() {
-        // need delete logic here.
+      classDelete() {
+        this.deleteClass(this.classid);
+        this.$emit('close');
```

Comment:
Been thinking about this a little bit, trying to come up with alternatives.

One option might be to just include a single vuex variable/mutation in the `pageState` like `modalShown` which takes constant values like `EDIT` and `CREATE`. There might be a few basic actions like `editClass` and `hideModal` to dispatch the necessary mutations.

Then we could move a lot of vuex state to local state `data` variables in the vue files: current modal values, errors, busy states, etc.

So in this scenario:
* we would still rely on a promise returned from the action to update local view state as you suggested
* since the 'modal visibility' state is controlled in vuex, the child component doesn't need to rely on `this.$emit('close');` to close itself because it would be able to call the `hideModal` action directly.
* The modal would be responsible for resetting its local data on mount.

thoughts?